src/features/world/scenes/BaseScene.ts

```diff
@@ -163,66 +232,15 @@ export class BaseScene extends Phaser.Scene {
     this.cursorKeys = this.input.keyboard?.createCursorKeys();
     this.input.keyboard?.removeCapture("SPACE");
 
-    console.log("initialiser room...");
-    // Respond to Websocket events
-    this.roomService.onEvent((event) => {
-      if (event.type === "CHAT_MESSAGE_RECEIVED") {
-        const { sessionId, text } = event as ChatMessageReceived;
-        if (
-          sessionId &&
-          String(sessionId).length > 4 &&
-          this.playerEntities[sessionId]
-        ) {
-          this.playerEntities[sessionId].speak(text);
-        }
-      }
-
-      if (event.type === "PLAYER_JOINED") {
-        const { sessionId, x, y, clothing } = event as PlayerJoined;
-        const player = this.createPlayer({
-          x,
-          y,
-          clothing,
-          isCurrentPlayer:
-            sessionId === this.roomService.state.context.room?.sessionId,
-        });
-        this.playerEntities[sessionId] = player;
-      }
-
-      if (event.type === "PLAYER_QUIT") {
-        const { sessionId } = event as PlayerQuit;
-        const entity = this.playerEntities[sessionId];
-        if (entity) {
-          entity.destroy();
-          delete this.playerEntities[sessionId];
-        }
-      }
-    });
-
-    this.roomService.onTransition((state) => {
-      if (state.value === "error" && !this.readonly) {
-        console.log("We have an error");
-        // Render the player for readonly
-        this.createPlayer({
-          x: 300,
-          y: 300,
-          isCurrentPlayer: true,
-          clothing: INITIAL_BUMPKIN.equipped,
-        });
-
-        this.readonly = true;
-      }
-    });
-    // If we cannot connect to server = render the player the anyway
-
-    // this.createPlayer({
-    //   x: 300,
-    //   y: 300,
-    //   isCurrentPlayer: true,
-    // });
+    this.roomService.off(this.eventListener);
+    this.roomService.off(this.transitionListener);
+    this.roomService.onEvent(this.eventListener);
+    this.roomService.onTransition(this.transitionListener);
```

Comment:
This removes listeners before adding new listeners. This is to attempt to prevent double listeners getting attached to the machine. This was causing double events.